Added ParticleSpeciesNames.md

DavidSagan opened the pull request:
This pull request has a new file for standardizing the names of particle species. See #176 .

Affected Components: None.
Writer/Reader/Data converter Changes: At this point none since the present standard does not reference particle species names.

DavidSagan committed:
Added ParticleSpeciesNames.md

ax3l commented on ParticleSpeciesNames.md:
```diff
@@ -0,0 +1,47 @@
+Names for Atoms and Molecules
+-------------------
+
+ - Atoms and molecules: Use standard chemical notation. Eg: "H20". Isotopes are denoted by a pound symbol "#" followed by the isotopic number followed by the chemical symbol. Eg: "#3He" for Helium-3.
```
maybe that's a stupid question, but is there some kind of ISO naming already for this on which we can rely on?

DavidSagan replied:
I looked but I could not find anything unfortunately.

ax3l replied:
I see, thanks for checking.
And something from PDG (the particle data group), maybe? Just that we don't miss something that already defines spellings.

ax3l replied:
(besides that insecurity of mine about accidentally re-inventing something, I like the proposed spellings)

DavidSagan replied:
I had looked at the PDG as well without luck.

ax3l reviewed (commented):
We will need to add:
- [ ] an extension ID `4` for 1.1.0 (because it will first be removed in 2.0.0)
- [ ] mention it in the base standard in the extensions section
- [ ] add the name  for the new `record` attribute
- [ ] define if this attribute is optional/recommended/required if the extension is declared for a file
- [ ] define if it is allowed to have more than one particle name per record, e.g. one could store a mesh with "summed charge density" of all ions and electrons -> list?

DavidSagan commented:
I realize I forgot to specify how to write atom and molecular charge states. So how about something like:

The charge state is denoted by a suffix with either:
1. Some number of plus or minus signs. Example: "H2O++"
1. A plus or minus sign with a following integer. Example: "CO+3"

ax3l commented:
I wonder if we want to record charge states really in the name.

We usually have them as a record in the species anyway (and for constant charge state as a constant record). Maybe other domains would do the same?

DavidSagan commented:
@ax3l: Having the charge state separately specified would be fine.

DavidSagan committed:
Update Draft of SpeciesType Extension

- file renamed to `EXT_SpeciesType.md`
- extension name: `SpeciesType` with extension-ID `4` (`1` is used and we reserved `2`$
- formatting unified
- defined `SpeciesType` as attribute: *recommended* if extension is declared
- allowed `lists`, e.g. for a local *summed charge density* in a non-equilibrium plasma
- atoms & isotopes: reference to IUPAC namings

ax3l committed:
EXT SpeciesType: Reference in Std

Reference to the new extension `SpeciesType` in the base standard

ax3l commented:
Changes I applied:
- file renamed to `EXT_SpeciesType.md`
- extension name: `SpeciesType` with extension-ID `4` (`1` is used and we reserved `2` in the past)
- formatting unified
- defined `SpeciesType` as attribute: *recommended* if extension is declared
- allowed `lists`, e.g. for a local *summed charge density* in a non-equilibrium plasma
- atoms & isotopes: reference to IUPAC namings
- reference in base standard

ax3l commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,80 @@
+
+- `SpeciesType`
+  - type: *(string)*
+  - scope: *recommended*
```
offline discussion: maybe better *optional* here, it's weird for e.g. "E fields" or arbitrary other records

Later on, in the `ED-PIC` extension or the `BeamPhysics` extension one can still specify, that it is *recommended* for e.g. `particle species`.

DavidSagan commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,80 @@
+- `SpeciesType`
+  - type: *(string)*
+  - scope: *recommended*
+  - description: particle species in this record, semicolon-separated list
```
Replace "semicolon-separated list" with "If there are multiple species to be specified, they can be specified using a semicolon separated list."

ax3l commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,80 @@
+
+This attribute can be used with any `record` (including `mesh records`).
+
+### SubAtomic Particle Names
```
Unify headline: SubAtomic Particles

DavidSagan commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,80 @@
+
+### Atoms & Isotopes
+
+Isotopes are denoted by a pound symbol `#` followed by the isotopic number followed by the chemical symbol, e.g.: `#3He` for Helium-3.
```
Switch order sentences. That is, start this section with "Element namings...".

ax3l commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,80 @@
+
+Isotopes are denoted by a pound symbol `#` followed by the isotopic number followed by the chemical symbol, e.g.: `#3He` for Helium-3.
+Element namings follow the abbreviated namings of the periodic table, defined by *The International Union of Pure and Applied Chemistry* (IUPAC).
+
```
Should we mention that the most abundant isotope can just be short-handed as `He` (for `#4He`)? It's somehow missing, but would allow ambiguity.

Is `D` ok for `#2H` or shall it be forced to the latter?

DavidSagan replied:
I would say that the standard should assume that  "He" is equivalent to "#4He".

DavidSagan replied:
Deuterium would be "#2H"

ax3l replied:
excellent, so we keep the abbreviations to a minimum: regular elements in the IUPAC periodic table

DavidSagan commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,80 @@
+
+### Molecules
+
+Use standard chemical notation, e.g.: `H20`.
```
The isotope prefix can be used with molecules. Example "#2H2O"  for two deuterium and "#2HHO" for one deuterium.

ax3l replied:
idea: we prefix the allowed values with the sections, e.g. `subatomic:`, `atom:`/`isotope:`, `molecule:`, ...

ax3l committed:
EXT SpeciesType: Update

- optional by default
- clarify annotations

ax3l commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,90 @@
+
+openPMD extension name: `SpeciesType`
+
+openPMD extension ID: `4`
```
note: this will be removed in `2.0.0` but we have to assign it since we are still in `1.X`

ax3l commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,90 @@
+
+### SubAtomic Particles
+
+  - `subatomic:antimuon`
```
offline discussion: we won't collide (yet) and if we should later one we can still add prefixes and convert known ones.

ax3l committed:
Remove Prefixes

Not needed (yet)

ax3l committed:
Fix Typos

DavidSagan committed:
Clarification Charge State

Add pointers

ax3l commented:
@RemiLehe we finished our proposal and you can now add your comments & review! :)

We tried to document all ideas and steps during our review today inline in comments, so if you like you can just un-collapse them for curiosity.

ax3l commented:
@RemiLehe ping :)

RemiLehe commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,94 @@
+  - `strange-quark`
+  - `tao-antiquark`
+  - `tao-quark`
+  - `tao-neutrino`
```
Shouldn't this be `tau-neutrino` (and adding `tau-antineutrino`) instead of `tao-neutrino` ? In addition, we should probably add `tau` as a particle (for the counterpart of `muon` and `electron`).

DavidSagan replied:
Yes. The change should be made.

RemiLehe commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,94 @@
+  - `muon-neutrino`
+  - `neutron`
+  - `photon`
+  - `pion`
```
Is there a reason to standardize the name of the `pion`, but not of all the other mesons?
(If the reason is that the `pion` is the most common, I am fine with this ; I am definitely not a specialist... I was just wondering.)

DavidSagan replied:
The list I came up with was meant to get the discussion going rather then  being complete (I am definitely not a specialist either). Do you have a list of names for the other mesons? By all means let us put them in.

RemiLehe commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,94 @@
+
+### SubAtomic Particles
+
+  - `antimuon`
```
Should we add `antitau`? (See remark below on the `tau` lepton.)

DavidSagan replied:
Yes we should add `antitau`. And `tao` is a typo. Should be `tau` everywhere.

RemiLehe commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,94 @@
+### SubAtomic Particles
+
+  - `antimuon`
+  - `antiproton`
```
Should we add `antineutron`?

DavidSagan replied:
Yes.

RemiLehe commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,94 @@
+
+  - `antimuon`
+  - `antiproton`
+  - `bottom`
```
Shouldn't this be `bottom-quark` and `bottom-antiquark` for consistency?

DavidSagan replied:
Yes.

RemiLehe commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,94 @@
+  - `bottom`
+  - `charm-antiquark`
+  - `charm-quark`
+  - `deuteron`
```
It seems that `deuteron` could be expressed as `#2H`. Using the convention below. Therefore, I suggest that we remove it from  this list.

DavidSagan replied:
Correct that `deuteron` is the same as `#2H` with a charge state of 1. In terms of having the name in the standard I don't care either way. @ax3l: Do you have an opinion?

ax3l replied:
sure, Deuterium should not be here

RemiLehe commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,94 @@
+  - `strange-antiquark`
+  - `strange-quark`
+  - `tao-antiquark`
+  - `tao-quark`
```
Also, I am not sure whether a `tao-quark` and `tao-antiquark` exist in the standard model... (If I remember well, the quarks of the standard model are `up`, `down`, `strange`, `charm`, `bottom`, `top`.)

DavidSagan replied:
Correct. That is a typo and should be removed.

RemiLehe commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,94 @@
+by the isotopic number followed by the chemical symbol, e.g.: `#3He`
+for Helium-3.
+
+The charge state is not encoded by the `SpeciesType` attribute.
```
By `charge state`, do we mean `ionization state`? Could `charge state` be considered ambiguous for some users?

ax3l replied:
Imho, charge state is the more general concept while ionization state only applies to atoms & molecules. This section just says we do not express it in this extension.

RemiLehe replied:
OK, sounds good.

DavidSagan replied:
I concur with Alex.

ax3l replied:
*Axel ;-)

RemiLehe commented:
Sorry for my late review! I added a couple of comments/questions.

RemiLehe commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,94 @@
+  - `down-antiquark`
+  - `down-quark`
+  - `electron`
+  - `electron-neutrino`
```
Should we add `electron-antineutrino`?

DavidSagan replied:
Sure.

RemiLehe commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,94 @@
+  - `graviton`
+  - `higgs-boson`
+  - `muon`
+  - `muon-neutrino`
```
Should we add `muon_antineutrino`?

DavidSagan replied:
Yes.

ax3l replied:
And I thought you wanted to point to Majorana neutrinos here ;)

ax3l commented:
Sorry, I totally forgot to swipe the subatomic particle list again in review.

I would suggest to make one list with elementary particles of the standard model (and their anti-partners), Gauge bosons and the higgs boson.

There is a plethora of namings for composite particles, especially Hadrons such as Mesons and Baryons, and should imho not be tried to be defined here, but instead reference e.g. the [particle data group](http://pdg.lbl.gov/2017/tables/contents_tables.html). Also, actual particle detector people think of jets and channels with particles which are again composites of the above. As far as we have no detector people here that suggest a naming I would also keep this free. Likely they would also use group identifiers #163

DavidSagan commented:
@ax3l 

>I would suggest to make one list with elementary particles of the standard model (and their anti-partners), Gauge bosons and the higgs boson.

Sounds good to me.

ax3l committed:
Elementary Particles & Gauge+Higgs Bosons

ax3l commented:
I updated the PR accordingly.

I would suggest to move this PR to `2.0.0` and close the `1.1.0` merge window so we do not merge it in last-minute and can take more time for it. Is that ok? I would then prepare the last clean-ups for `1.1.0` now.

ax3l commented on EXT_SpeciesType.md:
```diff
@@ -0,0 +1,95 @@
+
+Leptons:
+  - `electron` (`positron`)
+  - `electron-neutrino` (`anti-electron-neutrino`)
```
anti-neutrinos: who knows, maybe they turn out to be the same particles. but until now it seems that calling them anti-neutrinos is common

DavidSagan commented:
@ax3l: 
>I would suggest to move this PR to 2.0.0 and close the 1.1.0 merge window so we do not merge it in last-minute and can take more time for it. Is that ok? 

Fine with me.

RemiLehe commented:
I really like the reorganized version of the file. 
Thanks for

DavidSagan commented:
Yes I like the reorg too.
Minor stylistic suggestion: Use a comma instead of parenthesis. For example, Instead of:
  `electron ( positron )`
write:
  `electron, positron`

DavidSagan commented:
I know we don't want to in general get into trying to define names of mesons but thinking about it I would make an exception for `proton` and `anti-proton` since these are commonly used in experiments.

ax3l commented:
I am ok with both suggestions, just need to find a place to put `anti-proton` and maybe to say that `H` and `proton` are the same thing?

DavidSagan commented:
>I am ok with both suggestions, just need to find a place to put anti-proton and maybe to say that H and proton are the same thing? Actually they might not be the same, a user might want to refer to the element (atom) H...

It could be added that `#1H` is the same as `proton` if the charge is specified as `+1`.

ax3l commented:
I might just have been thinking loudly. `proton` is just more specific than `H` and that is fine, imho. No reason to explain more I think.

ax3l commented:
we will re-open this PR for further review into `upcoming-2.0.0`

ax3l commented:
Re-opened for 2.0.0 inclusion in #180
